Client/Frontend/Browser/Punycode.swift

```diff
@@ -0,0 +1,209 @@
+///* This Source Code Form is subject to the terms of the Mozilla Public
+// * License, v. 2.0. If a copy of the MPL was not distributed with this
+// * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+extension String {
+    private func toValue(index: Int) -> Character {
+        return [Character]("abcdefghijklmnopqrstuvwxyz0123456789".characters)[index]
```

Comment:
Extract `[Character]("abcdefghijklmnopqrstuvwxyz0123456789".characters)` to a top-level `private let` constant. That way, we don't need to recreate the character array every time these are called.